ibm_cloud_sdk_core/utils.py

```diff
@@ -117,6 +117,38 @@ def string_to_datetime(string: str) -> datetime.datetime:
     return val.replace(tzinfo=datetime.timezone.utc)
 
 
+def string_list_to_datetime_list(string_list: List[str]) -> List[datetime.datetime]:
+    """De-serializes each string in a list to a datetime.
+
+    Args:
+        string_list: list of strings containing datetime in iso8601 format.
+
+    Returns:
+        the de-serialized list of strings as a list of datetime objects.
+    """
+    if not isinstance(string_list, list):
+        return string_list
```

Comment:
I don't think we want to return the string list in case of an error do we?   The caller is expecting to get back a list of datetimes.  Could we throw an appropriate exception instead?

Reply:
Updated with a ValueError being raised for invalid argument types